app/controllers/api/subcollections/vms.rb

```diff
@@ -23,16 +23,31 @@ def vms_query_resource(object)
 
       private
 
+      def decorator_selection
+        params['decorators'].to_s.split(",")
+      end
+
+      def decorator_selection_for(collection)
+        decorator_selection.collect do |attr|
+          /\A#{collection}\.(?<name>.*)\z/.match(attr) { |m| m[:name] }
+        end.compact
+      end
+
       def create_vm_attributes_hash(vm_attrs, vm)
         vm_attrs.each_with_object({}) do |attr, hash|
           hash[attr] = vm.public_send(attr.to_sym) if vm.respond_to?(attr.to_sym)
         end.compact
       end
 
       def create_vm_decorators_hash(vm_decorators, vm)
-        vm_decorators.each_with_object({}) do |name, hash|
-          hash[name] = vm.decorate.public_send(name.to_sym) if vm.decorate.respond_to?(name.to_sym)
-        end.compact
+        hash = {}
+        if vm_decorators.include? 'supports_console?'
+          hash['supports_console?'] = vm.supports_console?
+        end
+        if vm_decorators.include? 'supports_cockpit?'
+          hash['supports_cockpit?'] = vm.supports_launch_cockpit?
+        end
+        hash
```

Comment:
Exactly :). The `attributes` version does not work today, because `SupportsFeatureMixin` does not create virtual attributes.

Not sure if the way forward is to make `SupportFeatureMixin` create virtual attributes for every feature, or to add special support for these in `attributes`, or to introduce a new query parameter..